File: Python/leetcode/1863.py
from typing import List


class Solution:
    def subsetXORSum(self, nums: List[int]) -> int:
        n = len(nums)
        res = 0
        for i in range(1 << n):
            xor_sum = 0
            for j in range(n):
                if i & (1 << j):
                    xor_sum ^= nums[j]
            res += xor_sum
        return res

File: Go/leetcode/1863.go
package Leetcode

func subsetXORSum(nums []int) int {
	return dfs(0, 0, nums)
}

// 递归
func dfs(val, idx int, nums []int) int {
	if idx == len(nums) { // 终止条件
		return val
	}
	// 选择当前元素
	return dfs(val^nums[idx], idx+1, nums) + dfs(val, idx+1, nums)
}
